### DISCUSSION TITLE
Blazor / SignalR - WebSocket connection to '...' failed: HTTP Authentication failed; no valid credentials available

### DISCUSSION THREAD

**Discussion by drma-tech** [dotnet/aspnetcore#23176] [General]
I'm using blazor (client) and set up a Hub (with Authorized attribute). Apparently it works, but this error is occurring. On the first day there was no error, but now it is. If I remove the attribute, the error does not occur.

My code looks like this:

```
hubConnection = new HubConnectionBuilder()
                .WithUrl(NavigationManager.ToAbsoluteUri("/chathub"), opt =>
                {
                    opt.AccessTokenProvider = () => Task.FromResult(token.Value);
                })
                .Build();
```

I need it that way, because in addition to the security issue, I need to send it to a specific user:

```
    [Authorize]
    public class ChatHub : Hub
    {
        public async Task SendPrivateMessage(string user, string message, string userId)
        {
            await Clients.User(userId).SendAsync("ReceivePrivateMessage", user, message);
        }
    }
```

**Comment by BrennanConroy** [comment]
Have you taken a look at the docs?
https://docs.microsoft.com/aspnet/core/signalr/authn-and-authz?view=aspnetcore-3.1#bearer-token-authentication

**Comment by drma-tech** [comment]
Yes, it was from that documentation that I got my code.

As I said, it works normal on localhost and it worked normal before on azure. This error has already happened in the past, but then I restarted the service (web app) and started working again, but now it doesn't work at all.

**Comment by drma-tech** [comment]
well, I can't even login to test it, since I have other errors:

[azure / blazor - SecurityException: Invalid assembly public key. (0x8013141E)](https://github.com/dotnet/aspnetcore/issues/22977)

**Comment by astrowalker** [comment]
Same story here -- in short if I remove `Authorize` attribute from the hub the access/auth token will be removed from the context (always) so I will lose the ability to check token in for example OnConnectedAsync -- so removing this attribute is no-go.

According to documentation I should add ConfigureJwtBearerOptions and plug it in. But using this class or not using it has the same effect -- first attempt is done over web sockets and it fails. On server side I can see the token is passed in query, it is good token, the next log is 
IdentityServer4.AccessTokenValidation.IdentityServerAuthenticationHandler: Information: AuthenticationScheme: Bearer was challenged.

and that's it. Apparently all subsequent calls avoid WS and basically it works, but this first error is troubling.
